client/consensus/manual-seal/src/rpc.rs

```diff
@@ -18,21 +18,30 @@
 
 //! RPC interface for the `ManualSeal` Engine.
 
-pub use self::gen_client::Client as ManualSealClient;
+use crate::error::{to_call_error, Error};
 use futures::{
 	channel::{mpsc, oneshot},
-	FutureExt, SinkExt, TryFutureExt,
+	FutureExt, SinkExt,
 };
-use jsonrpc_core::Error;
-use jsonrpc_derive::rpc;
+use jsonrpsee::types::{Error as JsonRpseeError, RpcModule};
 use sc_consensus::ImportedAux;
-use serde::{Deserialize, Serialize};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
 use sp_runtime::EncodedJustification;
 
-/// Future's type for jsonrpc
-type FutureResult<T> = Box<dyn jsonrpc_core::futures::Future<Item = T, Error = Error> + Send>;
-/// sender passed to the authorship task to report errors or successes.
-pub type Sender<T> = Option<oneshot::Sender<std::result::Result<T, crate::Error>>>;
+/// Helper macro to bail early in async context when you want to
+/// return `Box::pin(future::err(e))` once an error occurs.
+/// Because `Try` is not implemented for it.
+macro_rules! unwrap_or_fut_err {
```

Comment:
Nice

Reply:
I wonder if we should make use of this in other places as well?

Reply:
I have added it to the `rpc module` and duplicated it because I didn't want to add a dependency to it.

Reply:
but when we got the proc macros this can be removed